api/v1alpha1/hardwareclassification_types.go

```diff
@@ -216,6 +216,70 @@ const (
 	Empty ErrorType = ""
 )
 
+// MatchedCount will provide matched count of Hosts per profile
+type MatchedCount int
+
+// UnmatchedCount will provide unmatched count of Hosts per profile
+type UnmatchedCount int
+
+const (
+	// MatchedCount is the default status value
+	MatchedCountEmpty MatchedCount = 0
+	// MatchedCount is the default status value
+	UnmatchedCountEmpty UnmatchedCount = 0
+)
+
+// Total hosts in error state
+type ErrorHosts int
+
+// Total hosts in registration error state
+type RegistrationErrorHosts int
+
+// Total hosts in Introspection error state
+type IntrospectionErrorHosts int
+
+// Total hosts in Provisioning error state
+type ProvisioningErrorHosts int
+
+// Total hosts in Power Management error state
+type PowerMgmtErrorHosts int
+
+// Total hosts in Provisioned Registration Error state
+type ProvisionedRegistrationErrorHosts int
+
+// Total hosts in Preparation Error state
+type PreparationErrorHosts int
+
+// Total hosts in Detach Error state
+type DetachErrorHosts int
+
+const (
+	// ErrorHosts is count of Hosts in error state
+	ErrorHostsEmpty ErrorHosts = 0
+
+	// RegistrationErrorHosts is count of Hosts in Registration error state
+	RegistrationErrorHostsEmpty RegistrationErrorHosts = 0
+
+	// IntrospectionErrorHosts is count of Hosts in Introspection error state
+	IntrospectionErrorHostsEmpty IntrospectionErrorHosts = 0
+
+	// ProvisioningErrorHosts is count of Hosts in Provisioning error state
+	ProvisioningErrorHostsEmpty ProvisioningErrorHosts = 0
+
+	// PowerMgmtHosts is count of Hosts in Power Management error state
+	PowerMgmtHostsEmpty PowerMgmtErrorHosts = 0
+
+	// ProvisionedRegistrationErrorHosts is count of Hosts in Provisioned
+	// Registration error state
+	ProvisionedRegistrationErrorHostsEmpty ProvisionedRegistrationErrorHosts = 0
```

Comment:
do we really need this field. If host is provisioned, do you still see registration error ?

Reply:
"provisioned registration error" is a thing (i.e. it's separate to "registration error") that happens when we fail to adopt a host in ironic.

Reply:
@digambar15, yes we need it because BMH/metal3 have introduced multiple more error host fields. Now HWCC is supporting all erro host type provided by BMH. Please have a look https://github.com/metal3-io/baremetal-operator/blob/master/apis/metal3.io/v1alpha1/baremetalhost_types.go +137